### DISCUSSION TITLE
String Literals

### DISCUSSION THREAD

**Discussion by brk3** [cue-lang/cue#1801] [Q&A]
Hi, I'm new to cuelang but noticed these inconsistencies in the docs around string literals.

In the docs [here](https://cuelang.org/docs/tutorials/tour/types/stringlit/), it states:

> The whitespace directly preceding the closing quote must match the preceding whitespace on all other lines and is removed from these lines.

The actual behavior appears to be a bit more involved than this statement suggests:

Whitespace directly preceding the closing quote does *not* have to match the other lines, rather it must be *less than or equal to* to the other lines. The following works fine:

```
a: """
  hi, I'm indented by two spaces
"""
```

However this does not:

```
a: """
  hi, I'm indented by two spaces
    """ // three spaces
```

Finally, it seems that the leading whitespace stripping is based on the number of spaces that the final quote is indented by. E.g. the following will result in no leading whitespace:

```
a: """
  hi, I'm indented by two spaces
  """ // two spaces
```

The following will result in one character of whitespace:

```
a: """
  hi, I'm indented by two spaces
 """ // one space
```

**Comment by Lyoness** [comment]
Thanks for writing in and sorry we missed this. Hope you found an answer.

For future reference: The CUE [Github Discussions space](https://github.com/cue-lang/cue/discussions) is the place you can ask these types of questions, for the benefit of all who might have similar questions in future (and also to help improve documenation).

You can also chat about what you're trying to do and how CUE might help in the [community slack](https://join.slack.com/t/cuelang/shared_invite/enQtNzQwODc3NzYzNTA0LTAxNWQwZGU2YWFiOWFiOWQ4MjVjNGQ2ZTNlMmIxODc4MDVjMDg5YmIyOTMyMjQ2MTkzMTU5ZjA1OGE0OGE1NmE).

**Comment by Lyoness** [comment]
@brk3 Paul I've converted this issue to a discussion. Better late than never?  

Also will help future similar questions that may come (and improve docs).

**Comment by mvdan** [comment]
Thanks for spotting this - I agree that the current wording is confusing at best, when compared to the implementation. Perhaps @mpvl can shed some light.
